modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/asset/file_entry_full_content.jsp

```diff
@@ -44,6 +44,7 @@ DLViewFileVersionDisplayContext dlViewFileVersionDisplayContext = dlDisplayConte
 
 		<liferay-util:include page="/document_library/view_file_entry.jsp" servletContext="<%= application %>">
 			<liferay-util:param name="addPortletBreadcrumbEntries" value="<%= Boolean.FALSE.toString() %>" />
+			<liferay-util:param name="showBackButton" value="<%= Boolean.FALSE.toString() %>" />
```

Comment:
Instead of creating a new ad-hoc parameter for this, why not add the "portletResource" parameter to the URL generated from search (using the search portlet id as value), and checking if portletResource != DOCUMENT_LIBRARY_ADMIN when deciding when to display the navigation controls?

Reply:
I'm trying this but noticing that the back button is missing when viewing the document from Documents and Media. I assigned the portletResource inside `DLViewFileEntryDisplayContext.getRedirect()` but it looks like the url doesn't get built at that point in that case.

Would it work to do something like `boolean showBackButton = Objects.equals(dlRequestHelper.getResourcePortletName(), DLPortletKeys.DOCUMENT_LIBRARY_ADMIN);`  instead?